### PR TITLE
fix(ingestion/unity-catalog): fixed issue with profiling with GE turned on

### PR DESCRIPTION
**Pull request by dushayntAW**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved warehouse handling logic based on `include_hive_metastore` configuration, ensuring more accurate and reliable behavior.
  - Enhanced profiling process to better manage warehouse restarts, preventing potential issues during profiling.

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 1]
```diff
@@ -309,9 +309,19 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
             )
 
         if self.config.is_profiling_enabled():
-            self.report.report_ingestion_stage_start("Wait on warehouse")
-            assert wait_on_warehouse
-            wait_on_warehouse.result()
+            self.report.report_ingestion_stage_start("Start warehouse")
+            # Can take several minutes, so start now and wait later
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py:313]: This comment is not relevant here. May add a comment how the warehouse started earlier may turn off automatically by the time code reaches here and so we need to start warehouse again.

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 2]
```diff
@@ -309,9 +309,19 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
...
+            wait_on_warehouse = self.unity_catalog_api_proxy.start_warehouse()
+            if wait_on_warehouse is None:
+                self.report.report_failure(
+                    "initialization",
+                    f"SQL warehouse {self.config.profiling.warehouse_id} not found",
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py:318]: So this happens when permission for sql warehouse is missing Or incorrect sql warehouse is specified ?

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 3]
```diff
@@ -309,9 +309,19 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
...
+                )
+                return
+            else:
+                # wait until warehouse is started
+                wait_on_warehouse.result()
+            self.report.report_ingestion_stage_start("Warehouse started")
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py:324]: This line is not required. The next line starts a new ingestion stage, making this "Start warehouse" stage to automatically  stop.

**Review by mayurinehate — CHANGES_REQUESTED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py (hunk 1 above)
Comment (dushayntAW) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:313]: will update the comment

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py (hunk 2 above)
Comment (dushayntAW) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:318]: I need to check, as this was the existing codebase

Comment (dushayntAW) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:318]: this happens when there is no warehouse with specified id.

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:320]: makes sense

**Review by mayurinehate — APPROVED** [review]
LGTM

**Review by anshbansal — APPROVED** [review]

**Pull request merged**